substrate/client/network/src/litep2p/mod.rs

```diff
@@ -964,7 +964,7 @@ impl<B: BlockT + 'static, H: ExHashT> NetworkBackend<B, H> for Litep2pNetworkBac
 					Some(DiscoveryEvent::IncomingRecord { record: Record { key, value, publisher, expires }} ) => {
 						self.event_streams.send(Event::Dht(
 							DhtEvent::PutRecordRequest(
-								libp2p::kad::RecordKey::new(&key),
+								sc_network_types::kad::Key::new::<sc_network_types::kad::Key>(&key.to_vec().into()),
```

Comment:
Something like `RecordKey::from(key)` should work. In any case generic argument to new should not be used.